android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/views/overlays/PortraitNavigationOverlayView.kt

```diff
@@ -64,14 +68,35 @@ fun PortraitNavigationOverlayView(
 
     val cameraIsTrackingLocation = camera.value.state is CameraState.TrackingUserLocationWithBearing
 
+    val cameraControlState =
+        if (viewModel.isNavigating() && !cameraIsTrackingLocation) {
+          CameraControlState.ShowRecenter { camera.value = navigationCamera }
+        } else {
+          val bbox = uiState.routeGeometry?.boundingBox()
+          if (viewModel.isNavigating() && cameraIsTrackingLocation && bbox != null) {
+            CameraControlState.ShowRouteOverview {
+              val scale = density.density
+              camera.value =
+                  MapViewCamera.BoundingBox(
+                      LatLngBounds.from(bbox.north, bbox.east, bbox.south, bbox.west),
+                      padding =
+                          CameraPadding(20.0 * scale, 200.0 * scale, 100.0 * scale, 150.0 * scale))
```

Comment:
Pass in a data class of the instructions AND progress view sizes.

We also need to hide the road name!

Reply:
Scratch that; figured out something better :P Will comment up the new diff.